bigquery_etl/routine/parse_routine.py

```diff
@@ -14,40 +14,49 @@
 import yaml
 
 from bigquery_etl.metadata.parse_metadata import METADATA_FILE
-from bigquery_etl.util.common import project_dirs
 
 
-UDF_DIRS = ("udf", "udf_js")
-MOZFUN_DIR = ("sql/mozfun",)
 UDF_CHAR = "[a-zA-z0-9_]"
 UDF_FILE = "udf.sql"
 PROCEDURE_FILE = "stored_procedure.sql"
 EXAMPLE_DIR = "examples"
 TEMP_UDF_RE = re.compile(f"(?:udf|assert)_{UDF_CHAR}+")
-PERSISTENT_UDF_RE = re.compile(fr"((?:udf|assert){UDF_CHAR}*)\.({UDF_CHAR}+)")
-MOZFUN_UDF_RE = re.compile(fr"({UDF_CHAR}+)\.({UDF_CHAR}+)")
 PERSISTENT_UDF_PREFIX = re.compile(
     r"CREATE\s+(OR\s+REPLACE\s+)?FUNCTION(\s+IF\s+NOT\s+EXISTS)?", re.IGNORECASE
 )
+PERSISTENT_UDF_RE = re.compile(fr"((?:udf|assert){UDF_CHAR}*)\.({UDF_CHAR}+)")
 UDF_NAME_RE = re.compile(r"^([a-zA-Z0-9_]+\.)?[a-zA-Z][a-zA-Z0-9_]{0,255}$")
 GENERIC_DATASET = "_generic_dataset_"
+SQL_DIR = "sql/"
+ASSERT_UDF_DIR = "tests/assert"
 
-# UDFs defined in mozfun
-MOZFUN_ROUTINES = [
-    {
-        "name": root.split("/")[-2] + "." + root.split("/")[-1],
-        "is_udf": filename == UDF_FILE,
-    }
-    for udf_dir in MOZFUN_DIR
-    for root, dirs, files in os.walk(udf_dir)
-    for filename in files
-    if filename in (UDF_FILE, PROCEDURE_FILE)
-]
+
+def get_routines_from_dir(project_dir):
+    """Return all UDFs and stored procedures in the project directory."""
+    return [
+        {
+            "name": root.split("/")[-2] + "." + root.split("/")[-1],
+            "project": root.split("/")[-3],
+            "is_udf": filename == UDF_FILE,
+        }
+        for root, dirs, files in os.walk(project_dir)
+        for filename in files
+        if filename in (UDF_FILE, PROCEDURE_FILE)
+    ]
+
+
+def get_routines(project):
+    """Return all routines that could be referenced by the project."""
+    return (
+        get_routines_from_dir(project)
+        + get_routines_from_dir(os.path.join(SQL_DIR, "mozfun"))
```

Comment:
We should really be using pathlib for these directories

Reply:
(Note: I don't think we should block this PR from landing on this change, since the problem existed before it).